### DISCUSSION TITLE
Watcher to Node-RED | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1858] [Q&A]
# Watcher to Node-RED | Seeed Studio Wiki

Send message from Watcher to Node-Red

https://wiki.seeedstudio.com/watcher_to_node_red/

**Comment by eerke2021** [comment]
Method 1 requires cloud but method 2 is local?
Also I have 2 watchers. If I use method 2 is the “Chat ID” specific to each so I can tell which one it is?

**Comment by eerke2021** [comment]
trying Method 2. Nothing despite checking IP and port 3 times. Guessing this is because I have username/password?

**Comment by eerke2021** [comment]
method 1 works but there is about half a second between when watcher talks and node red updates. Don't think it's local

**Comment by eerke2021** [comment]
I said it works but it keeps forgetting my API Key when I redeploy.

**Comment by eerke2021** [comment]
hasn't forgotten Key last two deploys. What is the timeout on the server? The watcher can alarm twice locally in quick succession but only the first gets to Node Red. If you leave a big gap then both get through; 30s?

**Comment by eerke2021** [comment]
incidentally built in person detection and manual person; not a LLM mode.

**Comment by eerke2021** [comment]
disconnected again. This unreliability is reducing usefulness of method 1. All I am doing when I notice it stops working is copying the same API Key and it connects for a bit.
